### ISSUE TITLE
premature  gb termination over ZZ

### ISSUE THREAD

**Issue by jakobkroeker** [Macaulay2/M2#316]
```
R = ZZ[x, y, MonomialOrder => GRevLex]
I = ideal(-19*x*y+7*y^2-12*x,-5*y^3)
  gI = ideal gens gb I;
  assert( ( (gens I)%gI) == 0 );
 ggI = ideal gens gb gI;
 assert(numColumns (gens gI) == numColumns(gens ggI));
gI
ggI
```

output

```
i6 :  assert(numColumns (gens gI) == numColumns(gens ggI));
stdio:6:2:(3): error: assertion failed

i7 : gI

                 2                     2                2    3       3
o7 = ideal (5040y  - 8640x, x*y + 2387y  - 4092x, 34560x , 5y , 8640x )

o7 : Ideal of R

i8 : ggI

                 2                     2               2    3
o8 = ideal (5040y  - 8640x, x*y + 2387y  - 4092x, 8640x , 5y )

o8 : Ideal of R

```

**Comment by jakobkroeker** [comment]
less simpler failing example:
```
R = ZZ[x, y, MonomialOrder => GRevLex]
I = ideal(8*x^3+6,-22*x^3+8*x*y^2-17*y)
  gI = ideal gens gb I;
  assert( ( (gens I)%gI) == 0 );
 ggI = ideal gens gb gI;
 assert(numColumns (gens gI) == numColumns(gens ggI));
```
output
```
i6 :  assert(numColumns (gens gI) == numColumns(gens ggI));
stdio:6:2:(3): error: assertion failed

i7 : numColumns (gens gI)

o7 = 6

i8 : numColumns (gens ggI)

o8 = 9


```

**Comment by mikestillman** [comment]
Fixed in commit f701d87b89faf75.  This was a serious bug, sometimes giving incorrect answers.  The fix, though simple, might have efficiency repercussions, although there are larger examples that now run (correctly and) much faster.
